analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -287,20 +287,25 @@ operation.Kind switch
 
         private static ProgramState LearnBranchingConstraints(ControlFlowBranch branch, ProgramState state, IOperation branchValue)
         {
-            if (branchValue.Kind == OperationKindEx.Conversion)
-            {
-                var operand = IConversionOperationWrapper.FromOperation(branchValue).Operand;
-                state = LearnBranchingConstraints(branch, state, operand);
-                return state.SetOperationValue(branchValue, state[operand]);    // Propagate value from operand to conversion operation itself
-            }
-            else
-            {
-                var constraint = BoolConstraint.From((branch.Source.ConditionKind == ControlFlowConditionKind.WhenTrue) == branch.IsConditionalSuccessor);
-                state = state.SetOperationConstraint(branchValue, constraint);
-                return branchValue.TrackedSymbol() is { } symbol
-                    ? state.SetSymbolConstraint(symbol, constraint)
-                    : state;
-            }
+            var trueBranch = (branch.Source.ConditionKind == ControlFlowConditionKind.WhenTrue) == branch.IsConditionalSuccessor;
```

Comment:
What about `ControlFlowConditionKind.None`? If we don't expect it here, add a Debug.Assert.

Reply:
I'm not a big fan of those. `WhenTrue` and `IsConditionalSuccessors` makes enough sense together